src/include/units/physical/si/base/thermodynamic_temperature.h

```diff
@@ -42,17 +44,58 @@ struct kelvin_temperature_origin : point_origin<kelvin_temperature_origin, kelvi
 template<UnitOf<dim_thermodynamic_temperature> U = kelvin, QuantityValue Rep = double>
 using kelvin_temperature_point = quantity_point<dim_thermodynamic_temperature, U, Rep, kelvin_temperature_origin>;
 
+struct degree_celsius : alias_unit<kelvin, basic_symbol_text{"\u00b0C", "`C"}, no_prefix> {};
+
+struct celsius_temperature_origin : point_origin<celsius_temperature_origin, kelvin> {
+  using reference_origin = kelvin_temperature_origin;
+  static constexpr auto offset_to_reference = thermodynamic_temperature<millikelvin, long int>(273150);
+};
+
+template<UnitOf<dim_thermodynamic_temperature> U = degree_celsius, QuantityValue Rep = double>
+using celsius_temperature_point = quantity_point<si::dim_thermodynamic_temperature, U, Rep, celsius_temperature_origin>;
```

Comment:
Why allow specifying the unit if it's supposed to always be `degree_celsius`?

Reply:
Same as above, with `kelvin_temperature_point`.

Reply:
`millikelvin`?

Reply:
I was also thinking along the lines of allowing to specify the ratio. For example, for `celsius_temperature_point`, currently, you can specify an `UnitOf<dim_thermodynamic_temperature>`, like `kelvin`, which is not a celsius unit.

Reply:
`celsius_temperature_point<kelvin>` looks like lying, which is what I'm getting at. However, it might make sense. That's `quantity_point<si::dim_thermodynamic_temperature, kelvin, double, celsius_temperature_origin>`.

Reply:
Yeah, maybe only the ratio could be variable here....

Reply:
Yes. The problem is that the library only declares aliases for quantities of specific dimensions, but these are aliases for quantities of specific units. So they need their own way of declaration, which is to fix the unit (for `no_prefix`) or allow specifying the ratio (for `si::prefix`).
```suggestion
template<QuantityValue Rep = double>
using celsius_temperature_point = quantity_point<si::dim_thermodynamic_temperature, degree_celsius, Rep, celsius_temperature_origin>;
```

Reply:
Yes, we already discussed that probably for Celsius and Fahrenheit `no_prefix` should remain but we are aware of at least `millikelvin` so we should switch to `si::prefix` for kelvins.

Reply:
To me, `celsius_temperature_point<some_unit>` isn't even a thing it is just an alias to `quantity_point<dim_therm_temp, some_unit, some_origin>`. From a generic programming perspective, I don't see a point in restricting the combination between `some_unit` and `some_origin`. Each has a clear semantic meaning; one specifies the origin, the other the scaling of runtime values along the same "dimension". Preventing "unusual combinations" might prevent some errors at some points, but makes it generic programming more difficult, and worse, it creates confusion with respect to semantics of those two separate aspects. If we tie the origin to the unit, then both become names for the same thing and we lose semantics.

Reply:
Providing a general `thermodynamic_temperature_point` alias does make sense to me though. Then it's bike-shedding if we should provide either
```
kelvin_temperature_point<kelvin>
kelvin_temperature_point<millikelvin>
celsius_temperature_point<degree_celsius>
```
or
```
kelvin_temperature_point
millikelvin_temperature_point
celsius_temperature_point
```
or none at all.

Reply:
Keeping just a `thermodynamioc_temperature_point` is enough so I vote for "none at all" :-) Leave it to the users. They can always provide those by themselves if they need them. We can hint at such a usage in the example application.
Otherwise, we should provide the same for other dimensions to be consistent which would an overkill.

BTW, should we provide things like `speed_point` for other dimensions with the library?

Reply:
Are you sure? Please be aware, that currently `thermotdynamic_temperature_potin<>(0 * deg_C) == thermodynamic_temperature_point<>(0 * K)` (all with the same dimension-specific default origin). There is no such thing as a unit-specific default origin.

Reply:
I think this not true. The offset between the beginning of the scales for Celsius and Kelvin are well defined and the library should always use it when comparing or doing arithmetic over different units. Please read more in the review to the last commit.

Reply:
It is definitely true in the current implementation. I understand the desire for magic identification of origin points based on units to work, because in every-day use, that is what we do. This behaviour is currently implemented in `interpret_as_temperature_point`. But in everyday use, you never deal with expressions of temperatures, only concrete values.

To do magic identification of origin points by default, I think is very dangerous. It doesn't respect the semantics of usual physical quantities: When dealing with differences, `1 * K` and `1 * deg_C` are physically equivalent. If that behaviour is implemented, all of a sudden, the physical equivalence is erased: There is a way to express equivalent temperature points in Celsius and Kelvin, but the temperature points you get from equivalent temperature differences *will not be equivalent*.

The behaviour provided by the `quantity` type in this library (and for that matter most other unit libraries): It always represents differences, where the exact representation (i.e. unit) doesn't matter. All of a sudden, the equivalency is erased. That may look innocent with examples like the above, but consider the following:

 1) `thermodynamic_temperature_point( 1_q_deg_C * (1_q_km / 1_q_m))`: Should `unknown_unit` be forbidden?
 2) `thermodynamic_temperature_point( 1_q_Del)`: User defined the Delisle, so it's not `unknown_unit`, but she didn't tell the library what reference point to take. Should it be Kelvin? Should the library refuse?
 3) `quantity_point(1_q_ft)`: No specific reference point defined: Should the library refuse?
 4) `thermodynamic_temperature_point(1 * deg_C + 1 * deg_F)`
 5) `thermodynamic_temperature_point(1 * deg_F + 1 * deg_C)`

Note that there is no way for the library to refuse 4) and 5).

Reply:
In our library quantities are modeled as "relative". So yes, `1 * K` and `1 * deg_C` are physically equivalent. However, if you work on quantity points they should not be equivalent and I still hope it is possible to make it work with "identification of origin points by default" when casting between those units (because you have to cast in order to compare).

Reply:
However, you are right with your 1-5. Life is brutal :wink:.